src/main/kotlin/app/revanced/patches/shared/settings/preference/impl/SwitchPreference.kt

```diff
@@ -16,16 +13,19 @@ import org.w3c.dom.Element
  * @param default The default value of the switch.
  * @param summaryOn The summary to show when the preference is enabled.
  * @param summaryOff The summary to show when the preference is disabled.
+ * @param userDialogMessage The message to show in a dialog when the user toggles the preference.
  */
 internal class SwitchPreference(
     key: String, title: StringResource,
     val default: Boolean = false,
     val summaryOn: StringResource? = null,
-    val summaryOff: StringResource? = null
+    val summaryOff: StringResource? = null,
+    val userDialogMessage: StringResource? = null
 ) : BasePreference(key, title) {
     override val tag: String = "SwitchPreference"
 
     override fun serialize(ownerDocument: Document, resourceCallback: ((IResource) -> Unit)?): Element {
+        userDialogMessage?.include()
```

Comment:
While this works, out of consistency and convention reasons this `serialize` method should look like this:

```diff
 override fun serialize(ownerDocument: Document, resourceCallback: ((IResource) -> Unit)?): Element {
-        userDialogMessage?.include()
        return super.serialize(ownerDocument, resourceCallback).apply {
            addDefault(default)
            addSummary(summaryOn?.also { resourceCallback?.invoke(it) }, SummaryType.ON)
            addSummary(summaryOff?.also { resourceCallback?.invoke(it) }, SummaryType.OFF)
+           addUserDialogMessage(userDialogMessage?.also { resourceCallback?.invoke(it) }
        }
}
```

Reply:
That is a good point. What do you suggest? The [initial approach](https://github.com/revanced/revanced-patches/pull/1801/commits/f6f5efb76b1a071bde0c8f371a56de43e12dfe4d) seems a bit off to me as well since as you say, the string is not serialized into the XML switch preference.

Reply:
You noted that it is not part of the process of serializing the preference into XML though which I agree with. While it's easy to simply add it to `init` or to `serialize` both approaches don't seem right to me.

Reply:
That would lead to the initial issue that the string is meant for preference and not the patch, doesn't sound really right to me.

Reply:
The preference is meant for the patch. The string is meant for preference. This is a transitive dependency, for that reason the logic to add the string should be where the logic for the preference is.

Reply:
Alright, in that case, it should be fine as is for now.